subgraphs/compound-v3/src/sdk/token.ts

```diff
@@ -0,0 +1,267 @@
+    if (!symbolResultBytes.reverted) {
+      // for broken pairs that have no symbol function exposed
+      if (!this.isNullEthValue(symbolResultBytes.value.toHexString())) {
+        symbolValue = symbolResultBytes.value.toString();
+      } else {
+        // try with the static definition
+        const staticTokenDefinition =
+          StaticTokenDefinition.fromAddress(tokenAddress);
+        if (staticTokenDefinition != null) {
+          symbolValue = staticTokenDefinition.symbol;
+        }
+      }
+    }
+
+    return symbolValue;
+  }
+
+  private fetchTokenName(tokenAddress: Address): string {
+    const contract = ERC20.bind(tokenAddress);
+    const contractNameBytes = ERC20NameBytes.bind(tokenAddress);
+
+    // try types string and bytes32 for name
+    let nameValue = this.UNKNOWN_TOKEN_VALUE;
+    const nameResult = contract.try_name();
+    if (!nameResult.reverted) {
+      return nameResult.value;
+    }
+
+    // non-standard ERC20 implementation
+    const nameResultBytes = contractNameBytes.try_name();
+    if (!nameResultBytes.reverted) {
+      // for broken exchanges that have no name function exposed
+      if (!this.isNullEthValue(nameResultBytes.value.toHexString())) {
+        nameValue = nameResultBytes.value.toString();
+      } else {
+        // try with the static definition
+        const staticTokenDefinition =
+          StaticTokenDefinition.fromAddress(tokenAddress);
+        if (staticTokenDefinition != null) {
+          nameValue = staticTokenDefinition.name;
+        }
+      }
+    }
+
+    return nameValue;
+  }
+
+  private fetchTokenDecimals(tokenAddress: Address): i32 {
+    const contract = ERC20.bind(tokenAddress);
+
+    // try types uint8 for decimals
+    const decimalResult = contract.try_decimals();
+    if (!decimalResult.reverted) {
+      const decimalValue = decimalResult.value;
+      return decimalValue;
+    }
+
+    // try with the static definition
+    const staticTokenDefinition =
+      StaticTokenDefinition.fromAddress(tokenAddress);
+    if (staticTokenDefinition != null) {
+      return staticTokenDefinition.decimals as i32;
+    } else {
+      return this.INVALID_TOKEN_DECIMALS as i32;
+    }
+  }
+
+  private isNullEthValue(value: string): boolean {
+    return (
+      value ==
+      "0x0000000000000000000000000000000000000000000000000000000000000001"
+    );
+  }
+}
+
+// Initialize a Token Definition with the attributes
+export class StaticTokenDefinition {
+  address: Address;
+  symbol: string;
+  name: string;
+  decimals: i32;
+
+  // Initialize a Token Definition with its attributes
+  constructor(address: Address, symbol: string, name: string, decimals: i32) {
+    this.address = address;
+    this.symbol = symbol;
+    this.name = name;
+    this.decimals = decimals;
+  }
+
+  // Get all tokens with a static defintion
+  static getStaticDefinitions(): Array<StaticTokenDefinition> {
+    const staticDefinitions = new Array<StaticTokenDefinition>(6);
```

Comment:
Should this be outside the sdk? Since it is protocol specific.

Reply:
Maybe I was wrong, but I always thought these were atypical tokens on etherreum. And we used this in general, but it may make sense to remove:

ex: https://etherscan.io/address/0xbdeb4b83251fb146687fa19d1c660f99411eefe3#readContract

Reply:
Ah I see. It’s fine to leave them then 🤔 

Only thing is if we need to add additional ones for some protocols. We can either modify the lib to add that new token, or make it so that it is configurable, and allows to pass a list of other tokens. But we can leave it for now and work on that when we get there, if ever.